server/controllers/expenses.js

```diff
@@ -110,26 +110,33 @@ export const update = (req, res, next) => {
   const origExpense = req.expense;
   const newExpense = req.required.expense;
   const user = req.remoteUser || req.user;
-  const modifiableProps = [
-    'amount',
-    'attachment',
-    'category',
-    'comment',
-    'incurredAt',
-    'currency',
-    'notes',
+  const modifiablePropsWhenStatusApproved = [
     'payoutMethod',
+    'category',
     'title',
-    'vat'
+    'incurredAt',
+    'notes'
   ];
 
-  assertExpenseStatus(origExpense, status.PENDING)
-    .tap(() => {
-      modifiableProps.forEach(prop => origExpense[prop] = newExpense[prop] || origExpense[prop]);
-      origExpense.updatedAt = new Date();
-      origExpense.lastEditedById = user.id;
-    })
-    .then(() => origExpense.save())
+  // following fields can only be modified while an expense is PENDING
+  const modifiablePropsWhenStatusPending = modifiablePropsWhenStatusApproved.concat([
+    'amount',
+    'currency',
+    'vat',
+    'attachment'
+  ]);
+
```

Comment:
This is code writing style, but I like to put these like this:

```
const editableProps = {
  "approved": [attr1, attr2, ...]
  "pending": [attr1, attr2, attr3, attr4, ..]
}
```

Or if you really want to concat:

```
const editableProps = {}
editableProps.approved = [attr1 ,attr2, ...];
editableProps.pending = [...editableProps.approved, attr3, attr4, ...];
```